ddtrace/contrib/tornado/__init__.py

```diff
@@ -0,0 +1,73 @@
+"""
+The Tornado integration traces all ``RequestHandler`` defined in a Tornado web application.
+Auto instrumentation is available using the ``trace_app`` function as follows::
+
+    import tornado.web
+    import tornado.gen
+    import tornado.ioloop
+
+    from ddtrace import tracer
+    from ddtrace.contrib.tornado import trace_app
+
+    # create your handlers
+    class MainHandler(tornado.web.RequestHandler):
+        @tornado.gen.coroutine
+        def get(self):
+            self.write("Hello, world")
+
+    # create your application
+    app = tornado.web.Application([
+        (r'/', MainHandler),
+    ])
+
+    # trace your application before the execution
+    trace_app(app, tracer, service='tornado-site')
+
+    # and run it as usual
+    app.listen(8888)
+    tornado.ioloop.IOLoop.current().start()
+
+When a ``RequestHandler`` is hit, a request span is automatically created and attached
+to the current ``request`` object, so that it can be used in the application code::
+
+    class MainHandler(tornado.web.RequestHandler):
+        @tornado.gen.coroutine
+        def get(self):
+            ctx = getattr(self.request, 'datadog_context')
+            # do something with the tracing Context
+
+If you want to trace other part of your application, you can use both the ``Tracer.wrap()``
+decorator and the ``Tracer.trace()`` method::
+
+    class MainHandler(tornado.web.RequestHandler):
+        @tornado.gen.coroutine
+        def get(self):
+            yield self.notify()
+            with tracer.trace('tornado.post_notify') as span:
+                # do more work
+
+        @tracer.wrap('tornado.notify', service='tornado-notification')
```

Comment:
actually it should time correctly the function because we're not using the default `@tracer.wrap` but a special wrapper [here][1]. This means that the API is the same as before, but in the case of Tornado we automatically handle the Future monitoring.

We may need to discuss a little bit more on this functionality but the idea is that the API should remain the same.

[1]: https://github.com/DataDog/dd-trace-py/pull/204/files#diff-3b957a6da0afb54a68360c040397e07aR21